File: AD-HOC/1802/main.py
def get_prices(v_p, v_m, v_f, v_q, v_b):
    prices = []
    for p in v_p:
        for m in v_m:
            for f in v_f:
                for q in v_q:
                    for b in v_b:
                        prices.append(p + m + f + q + b)
    return prices


def sum_most_expensive(k, prices):
    return sum(sorted(prices, reverse=True)[0:k])


_, *v_p = map(int, input().split())
_, *v_m = map(int, input().split())
_, *v_f = map(int, input().split())
_, *v_q = map(int, input().split())
_, *v_b = map(int, input().split())
k = int(input())
prices = get_prices(v_p, v_m, v_f, v_q, v_b)
print(sum_most_expensive(k, prices))

File: AD-HOC/1802/main.rb
def get_prices(v_p, v_m, v_f, v_q, v_b)
  prices = []
  v_p.each do |p|
    v_m.each do |m|
      v_f.each do |f|
        v_q.each do |q|
          v_b.each do |b|
            prices.push(p + m + f + q + b)
          end
        end
      end
    end
  end
  prices
end

def sum_most_expensive(k, prices)
  prices.sort.reverse[...k].sum
end

_, *v_p = gets.split.map(&:to_i)
_, *v_m = gets.split.map(&:to_i)
_, *v_f = gets.split.map(&:to_i)
_, *v_q = gets.split.map(&:to_i)
_, *v_b = gets.split.map(&:to_i)
k = gets.to_i
prices = get_prices(v_p, v_m, v_f, v_q, v_b)
puts sum_most_expensive(k, prices)
